### PR TITLE
Merge v1.4.20 into 1.5

### PR DESCRIPTION
**Pull request by phearnot**

### CODE REVIEW

File: lang/tests-js/src/test/scala/com/wavesplatform/test/builtInFunctions/list/worksheet.sc [hunk 1]
```diff
@@ -0,0 +1 @@
+167
```
Comment (vsuharnikov) [inline lang/tests-js/src/test/scala/com/wavesplatform/test/builtInFunctions/list/worksheet.sc]: Is this file required?

File: node/src/main/scala/com/wavesplatform/api/http/utils/UtilsEvaluator.scala [hunk 2]
```diff
@@ -12,105 +14,136 @@ import com.wavesplatform.lang.contract.DApp
...
-          ContractLimits.MaxCallableActionsAmountBeforeV6(script.stdLibVersion),
-          ContractLimits.MaxBalanceScriptActionsAmountV6,
-          ContractLimits.MaxAssetScriptActionsAmountV6,
-          ContractLimits.MaxWriteSetSize,
-          ContractLimits.MaxTotalWriteSetSizeInBytes
-        ),
-        availablePayments = ContractLimits.MaxTotalPaymentAmountRideV6,
-        currentSnapshot = paymentsSnapshot,
-        invocationRoot = DAppEnvironment.InvocationTreeTracker(DAppEnvironment.DAppInvocation(dAppAddress, null, Nil))
+        Map()//Map(invoke.sender.toAddress -> Portfolio.build(invoke.feeAssetId -> -invoke.fee))
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/api/http/utils/UtilsEvaluator.scala]: 🤔

File: node/src/test/scala/com/wavesplatform/state/BlockchainUpdaterImplSpec.scala [hunk 3]
```diff
@@ -86,7 +86,7 @@ class BlockchainUpdaterImplSpec extends FreeSpec with EitherMatchers with WithDo
     (master, List(genesisBlock, b1, b2))
   }
 
-  "blockchain update events sending" - {
+  "blockhain update events sending" - {
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/BlockchainUpdaterImplSpec.scala]: Typo

**Review by ivan-mashonskiy — APPROVED** [review]

**Review by vsuharnikov — APPROVED** [review]

**Review by xrtm000 — APPROVED** [review]
